src/prommis/nanofiltration/nf_brine.py

```diff
@@ -0,0 +1,386 @@
+####################################################################
+# This code has been adapted from a test in the WaterTAP repo to
+# simulate a simple separation of lithium and magnesium
+#
+# watertap > unit_models > tests > test_nanofiltration_DSPMDE_0D.py
+# test defined as test_pressure_recovery_step_2_ions()
+#
+# also used the following flowsheet as a reference
+# watertap > examples > flowsheets > nf_dspmde > nf.py
+#
+# https://github.com/watertap-org/watertap/blob/main/tutorials/nawi_spring_meeting2023.ipynb
+####################################################################
+
+"""
+Nanofiltration flowsheet for Donnan steric pore model with dielectric exclusion
+"""
+
+# import statements
+from pyomo.environ import (
+    ConcreteModel,
+    Constraint,
+    Objective,
+    TransformationFactory,
+    floor,
+    log10,
+)
+from pyomo.network import Arc
+
+import idaes.core.util.scaling as iscale
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers import get_solver
+from idaes.core.util.initialization import propagate_state
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.models.unit_models import Feed, Product
+import idaes.logger as idaeslog
+
+from watertap.property_models.multicomp_aq_sol_prop_pack import (
+    ActivityCoefficientModel,
+    DensityCalculation,
+    MCASParameterBlock,
+)
+from watertap.unit_models.nanofiltration_DSPMDE_0D import NanofiltrationDSPMDE0D
+from watertap.unit_models.pressure_changer import Pump
+
+_log = idaeslog.getLogger(__name__)
+
+
+def main():
+    """
+    Builds and solves the NF flowsheet
+
+    Returns:
+        m: pyomo model
+    """
+    solver = get_solver()
+    m = build()
+
+    initialize(m, solver)
+    _log.info("Initialization Okay")
+
+    if degrees_of_freedom(m) != 0:
+        raise Exception("Degrees of freedom were not equal to zero")
```

Comment:
It is generally considered bad practice to use a bare `Exception`, as it makes it harder to differentiate different types of failures (`except Exception:` catches basically everything). You should use more specific exception types wherever possible; here I suggest `ValueError`.